test/functional/luacats_grammar_spec.lua

```diff
@@ -0,0 +1,108 @@
+local helpers = require('test.functional.helpers')(after_each)
+local eq = helpers.eq
+
+local grammar = require('src/nvim/generators/luacats_grammar')
+
+describe('luacats grammar', function()
+  --- @param text string
+  --- @param exp table<string,string>
+  local function test(text, exp)
+    it(string.format('can parse %q', text), function()
+      eq(exp, grammar:match(text))
+    end)
+  end
+
+  test('@param hello vim.type', {
+    name = 'hello',
+    type = 'vim.type',
+  })
+
+  test('@param hello vim.type this is a description', {
+    name = 'hello',
+    type = 'vim.type',
+    desc = 'this is a description'
+  })
+
+  test('@param hello vim.type|string this is a description', {
+    name = 'hello',
+    type = 'vim.type|string',
+    desc = 'this is a description'
+  })
+
+  test('@param hello vim.type?|string? this is a description', {
+    name = 'hello',
+    type = 'vim.type?|string?',
+    desc = 'this is a description'
+  })
+
+  test('@return string hello this is a description', {
+    name = 'hello',
+    type = 'string',
+    desc = 'this is a description'
+  })
+
+  test('@return fun() hello this is a description', {
+    name = 'hello',
+    type = 'fun()',
+    desc = 'this is a description'
+  })
+
+  test('@return fun(a: string[]): string hello this is a description', {
+    name = 'hello',
+    type = 'fun(a: string[]): string',
+    desc = 'this is a description'
+  })
+
+  test('@return fun(a: table<string,any>): string hello this is a description', {
+    name = 'hello',
+    type = 'fun(a: table<string,any>): string',
+    desc = 'this is a description'
+  })
+
+  test('@param ... string desc', {
+    name = '...',
+    type = 'string',
+    desc = 'desc'
+  })
+
+  test('@param level (integer|string) desc', {
+    name = 'level',
+    type = '(integer|string)',
+    desc = 'desc'
+  })
+
+  test('@return (string command) the command and arguments', {
+    name = 'command',
+    type = 'string',
+    desc = 'the command and arguments'
+  })
+
+  test('@return (string command, string[] args) the command and arguments', {
```

Comment:
I think multiple (variable) returns is like: `@return string...\n@return string[]...`

https://luals.github.io/wiki/annotations/#return

Reply:
It's both:

```lua
@return string a, string b
```

is equivalent to:
```
@return string a
@return string b
```

Reply:
Oh didn't know that. TIL, thanks! I'm gonna fix the luadoc treesitter parser.